### PR TITLE
Fix depwarns on 0.5

### PR DESCRIPTION
**Pull request by nkottary**
Fix deprecation warnings on julia v0.5

### CODE REVIEW

File: src/Blosc.jl [hunk 1]
```diff
@@ -161,16 +161,26 @@ end
 # Allow Blosc to set the optimal blocksize (default)
 set_default_blocksize() = set_blocksize(0)
 
+if VERSION < v"0.5-"
+    _bytestring = bytestring
+else
+    _bytestring = unsafe_string
+end
```
Comment (stevengj) [inline src/Blosc.jl]: This shouldn't be necessary.   If you update the REQUIRE file to list `Compat 0.8`, then you can just call `unsafe_string` (which is defined in older versions by Compat).

File: src/Blosc.jl [hunk 2]
```diff
@@ -161,16 +161,26 @@ end
...
 end
 
 immutable CompressionInfo
-    library::ByteString
+    @static if VERSION < v"0.5-"
+        library::ByteString
+    else
+        library::String
+    end
     typesize::Int
```
Comment (stevengj) [inline src/Blosc.jl]: Can just use `Compat.String` here.

File: src/Blosc.jl [hunk 3]
```diff
@@ -200,8 +210,8 @@ function compressor_info(name::AbstractString)
                 (Cstring,Ptr{Ptr{Cchar}},Ptr{Ptr{Cchar}}),
                 name, lib, ver)
     ret < 0 && error("Error retrieving compressor info for $name")
-    lib_str = bytestring(lib[1]); Libc.free(lib[1])
-    ver_str = bytestring(ver[1]); Libc.free(ver[1])
+    lib_str = _bytestring(lib[1]); Libc.free(lib[1])
```
Comment (stevengj) [inline src/Blosc.jl]: These can be updated to `lib_str = unsafe_wrap(Compat.String, lib[1], true)`, which eliminates the need to call `free`.

**Comment by nkottary** [comment]
Updated the code as per second comment above.

I was not able to use `unsafe_string` or `unsafe_wrap` on v0.4 (comments 1 and 3 above).

I get the following error on running `Pkg.test`:
```
ERROR: LoadError: MethodError: `unsafe_string` has no method matching unsafe_string(::Ptr{Int8})
 in anonymous at no file
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:320
 in process_options at ./client.jl:280
 in _start at ./client.jl:378
while loading /home/nishanth/.julia/v0.4/Blosc/test/runtests.jl, in expression starting on line 11
```

**Comment by nkottary** [comment]
Updated the code as per above suggestion.

File: test/runtests.jl [hunk 4]
```diff
@@ -19,8 +20,8 @@ Blosc.set_blocksize(0)
 Blosc.set_default_blocksize()
 @test_throws ArgumentError Blosc.set_blocksize(-1)
 
-s = convert(ASCIIString, rand('0':'z', 10000))
-@test ASCIIString(decompress(UInt8, compress(s))) == s
+s = convert(String, rand('0':'z', 10000))
+@test String(decompress(UInt8, compress(s))) == s
```
Comment (stevengj) [inline test/runtests.jl]: Should be `Compat.ASCIIString`.

**Comment by timholy** [comment]
@stevengj, can you tag a new release?

**Comment by stevengj** [comment]
Will do.

**Pull request merged**